library/src/main/java/com/bumptech/glide/Registry.java

```diff
@@ -69,7 +71,23 @@ public Registry() {
    * to replace Glide's default {@link Encoder}s.
    */
   public <Data> Registry register(Class<Data> dataClass, Encoder<Data> encoder) {
-    encoderRegistry.add(dataClass, encoder);
+    encoderRegistry.append(dataClass, encoder);
+    return this;
+  }
+
+  /**
+   * Prepends the given {@link Encoder} into the list of available {@link Encoder}s
+   * so that it is attempted before all later and default {@link Encoder}s for the given
+   * data class.
+   *
+   * <p>This method allows you to replace the default {@link Encoder} because it ensures
+   * the registered {@link Encoder} will run first. You can return false in
```

Comment:
Returning false will cause the load to fail. There isn't currently any logic that tries Encoders in order, it just assumes there's one per data type.

Reply:
https://github.com/bumptech/glide/blob/35c28c28ca2a030e90df5aac67b0356c769cd623/library/src/main/java/com/bumptech/glide/load/engine/SourceGenerator.java#L74